core/src/main/java/org/infinispan/commands/functional/ManyDataCommand.java

```diff
@@ -0,0 +1,10 @@
+package org.infinispan.commands.functional;
+
+import java.util.Set;
+
+/**
+ * @author Radim Vansa &lt;rvansa@redhat.com&gt;
+ */
+public interface ManyDataCommand<K> {
```

Comment:
I don't really see the need to add yet another interface to the huge chain of *Command of interfaces we already have. What I mean by that is this: Each AbstractWriteManyCommand implementation could have a getKeys() method inside, that does not depend on an interface, and when it arrives to the interceptor level, each instance of the command, for which we have a visit* call, could call its own getKeys() and pass that to the method.

In fact, I think adding this interface makes the code more complex unnecessarily by having to use `&` structures to find commands that are X and Y...etc.

Truth to be said, I did the same error with ParamsCommand, and I can see now how this could get out for unnecessary reasons really.